### ISSUE TITLE
XIOS hangs or crashes with multi-file output

### ISSUE THREAD

**Issue by hiker** [hiker/xios-2252#3]
If (at least one) xios server has no data points, the following test in nc4_data_output will cause these servers to return prematurely, not taking part in an MPI global operation later:
```
         if (domain->isEmpty())
           if (SuperClass::type==MULTI_FILE) return;
```
All processes that have data will reach a global operation (even though MULTI_FILE is set):
```
xios :: CFile :: checkWriteFile (file.cpp:330)
xios :: CFile :: createHeader (file.cpp:592)
xios :: CDataOutput :: writeFieldGrid (data_output.cpp:118)
xios :: CDataOutput :: writeGrid (data_output.cpp:18)
xios :: CDataOutput :: writeGrid (data_output.cpp:72)
xios :: CNc4DataOutput :: writeDomain (nc4_data_output.cpp:61)
xios :: CDomain :: computeWrittenCompressedIndex (domain.cpp:2229)
PMPI_Allreduce (pallreduce.c:113)
...
```
This likely results in crashes (or hanging application, or incorrectly mixing of data, because the prematurely returned process might continue writing other fields, till it reaches a different field for which it has data, and then the MPI_Allreduce will be done over different fields, see file.cpp:592 for the loop over all fields).

In my test case it creates a NetCDF exception in nc4_data_output, around line 2197:
```
       catch (CNetCdfException& e)
         {
           StdString msg("On writing field : ");
           msg.append(fieldid); msg.append("\n");
           msg.append("In the context : ");
           msg.append(context->getId()); msg.append("\n");
           msg.append(e.what());
           ERROR("CNc4DataOutput::writeField_(CField* field)", << msg);
         }
      } // writeField_()
```
with the message `"On writing field : simsk\nIn the context : nemo_server\nError when calling function nc_inq_dimid(ncid, dimName.c_str(), &dimId)\nNetCDF: Invalid dimension ID or name\nUnable to get id of dimension, given "...`. Removing the first mention tests fixes this issue.

**Comment by hiker** [comment]
Reason for the NetCDF error is likely that the code skipped because of the early return statement in case of an empty domain is responsible for writing dimension information, which is then missing later.

**Comment by hiker** [comment]
Closed with #4.
